src/Discord.Net/Entities/Users/IMember.cs

```diff
@@ -0,0 +1,38 @@
+// todo: meth
+// todo: prop
+// todo: docs
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Wumpus.Entities;
+
+namespace Discord
+{
+    public interface IMember : ISnowflakeEntity, IVoiceState
+    {
+        ValueTask<IUser> GetUserAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        ValueTask<IGuild> GetGuildAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        ValueTask<IReadOnlyCollection<IRole>> GetRoles(StateBehavior stateBehavior = default, RequestOptions? options = default);
+
+        ulong GuildId { get; }
+        IReadOnlyCollection<ulong> RoleIds { get; }
+        IMemberPresence Presence { get; }
+
+        DateTimeOffset? JoinedAt { get; }
+        string Nickname { get; }
+        GuildPermissions GuildPermissions { get; }
+
+        ValueTask<ChannelPermissions> GetPermissionsAsync(IGuildChannel channel);
+
+        Task ModifyAsync(MemberProperties properties, RequestOptions? options = default);
+
+        // review: should these two be extensions on IGuild?
```

Comment:
There should at least be an id version of it on IGuild.

Reply:
That thought should apply to almost all API functionality. Except, perhaps, when there should be a "does this ID exist" check, in which case performing the `Get()` call serves that purpose.